src/main/java/hudson/plugins/active_directory/ActiveDirectoryUserDetail.java

```diff
@@ -117,27 +115,34 @@ public int hashCode() {
 
     @Override
     protected void setAuthorities(GrantedAuthority[] authorities) {
-        SecurityRealm realm = Jenkins.getInstance().getSecurityRealm();
-        if ((realm instanceof ActiveDirectorySecurityRealm)) {
-            ActiveDirectorySecurityRealm activeDirectoryRealm = (ActiveDirectorySecurityRealm)realm;
-            if (activeDirectoryRealm.removeIrrelevantGroups) {
-                Set<String> referencedGroups = new HashSet<String>();
-                for (String group : Jenkins.getInstance().getAuthorizationStrategy().getGroups()) {
-                    referencedGroups.add(group.toLowerCase());
-                }
-                // We remove irrelevant groups only if the active AuthorizationStrategy has any referenced groups:
-                if (!referencedGroups.isEmpty()) {
-                    List<GrantedAuthority> relevantGroups = new ArrayList<GrantedAuthority>();
-
-                    for (GrantedAuthority group : authorities) {
-                        String groupName = group.getAuthority();
-                        if (groupName != null && referencedGroups.contains(groupName.toLowerCase())) {
-                            relevantGroups.add(group);
-                        }
-                    }
-                    authorities = relevantGroups.toArray(new GrantedAuthority[relevantGroups.size()]);
-                }
-            }
+        final Jenkins instance = Jenkins.getInstance();
+        if(instance!=null) {
```

Comment:
Depending on what to expect, you may be interested in using Jenkins.getActiveInstance() instead to remove the null check above